### ISSUE TITLE
'Jest did not exit one second after the test run has completed.' warning when using <Observer>

### ISSUE THREAD

**Issue by bscaspar** [mobxjs/mobx-react-lite#288]
## Intended outcome
Jest runs and finishes when expected.

## Actual outcome
After adding an `<Observer>` to a component and using that component in a test with Jest and react-test-renderer, Jest runs successfully but returns a warning `Jest did not exit one second after the test run has completed.`

## How to reproduce the issue
[Repo](https://github.com/bscaspar/jest-observer-test-demo/tree/master)
clone, `yarn`, `jest`

**Comment by alanhr** [comment]
hello, @FredyC .

I'm face the same problem. Do you have any update about that ?

**Comment by alanhr** [comment]
Hello, @FredyC .

I'm facing the same problem. Do you have any update about that?

**Comment by FredyC** [comment]
@alanhr Not really, if you read my last comment, there is no clear path on how to proceed here. React is not giving proper tools for handling this.

**Comment by mweststrate** [comment]
Hmm, it would be interesting to see how recoil or relay handle this, I'm
pretty sure they have a similar clean up strategy.

**Comment by liamkernighan** [comment]
`jest.useFakeTimers()` is solving the problem.

**Comment by alanhr** [comment]
@liamkernighan this was my solution to solve that problem

**Comment by FredyC** [comment]
@liamkernighan It's cool, but it's more like a workaround. Faking timers can have unforeseen consequences. It's easy to forget about it and then be surprised why is test stuck.

**Comment by RoystonS** [comment]
It would be straightforward to expose a 'debug-only' call (for unit-testing use only) that shuts down the timer, and call that from the Jest (or other unit test framework) teardown.

**Comment by mweststrate** [comment]
Anyone feel free to PR a `clearTimers` utility. 

P.S. I noticed that in the above example the test example the render tree is never cleaned up, so doing that might help as well.
